precompiles/werc20/tx.go

```diff
@@ -0,0 +1,84 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package werc20
+
+import (
+	"fmt"
+	"math/big"
+
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	cmn "github.com/evmos/evmos/v20/precompiles/common"
+	"github.com/evmos/evmos/v20/x/evm/core/vm"
+	evmtypes "github.com/evmos/evmos/v20/x/evm/types"
+)
+
+const (
+	// DepositMethod defines the ABI method name for the IWERC20 deposit
+	// transaction.
+	DepositMethod = "deposit"
+	// WithdrawMethod defines the ABI method name for the IWERC20 withdraw
+	// transaction.
+	WithdrawMethod = "withdraw"
+)
+
+// Deposit handles the payable deposit function. It retrieves the deposited amount
+// and sends it back to the sender using the bank keeper.
+func (p Precompile) Deposit(ctx sdk.Context, contract *vm.Contract, stateDB vm.StateDB) ([]byte, error) {
+	caller := contract.Caller()
+	depositedAmount := contract.Value()
+
+	callerAccAddress := sdk.AccAddress(caller.Bytes())
+	precompileAccAddr := sdk.AccAddress(p.Address().Bytes())
+
+	// Send the coins back to the sender
+	if err := p.BankKeeper.SendCoins(
+		ctx,
+		precompileAccAddr,
+		callerAccAddress,
+		sdk.NewCoins(sdk.Coin{
+			Denom:  evmtypes.GetEVMCoinDenom(),
+			Amount: math.NewIntFromBigInt(depositedAmount),
+		}),
+	); err != nil {
+		return nil, err
+	}
+
+	// Add the entries to the statedb journal in 18 decimals.
+	convertedAmount := evmtypes.ConvertAmountTo18DecimalsBigInt(depositedAmount)
```

Comment:
the amount is the msg.value. So it is already in 18 decimals

Reply:
Good catch!